ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/TransactionsWriter.scala

```diff
@@ -117,107 +116,172 @@ private[dao] final class TransactionsWriter(
         .foldLeft[WitnessRelation[ContractId]](Map.empty)(Relation.merge)
     }
 
-  private def prepareWitnessesBatch(
+  private def fullDivulgence(
       toBeInserted: Set[ContractId],
       toBeDeleted: Set[ContractId],
       transient: Set[ContractId],
       transaction: Transaction,
       blinding: BlindingInfo,
-  ): Option[BatchSql] = {
+  ): WitnessRelation[ContractId] = {
     val localDivulgence = divulgence(transaction, blinding.disclosure, toBeInserted)
     val fullDivulgence = Relation.union(
       localDivulgence,
       blinding.globalDivulgence.filterKeys(cid => !toBeDeleted(cid) && !transient(cid))
     )
-    val insertWitnessesBatch = contractWitnessesTable.prepareBatchInsert(fullDivulgence)
-    if (localDivulgence.nonEmpty) {
-      assert(insertWitnessesBatch.nonEmpty, "No witness found for contracts marked for insertion")
-    }
-    insertWitnessesBatch
+    fullDivulgence
```

Comment:
You can probably just remove the binding and return the result of the previous expression.